src/java.base/share/classes/sun/security/ec/ed/EdDSAPrivateKeyImpl.java

```diff
@@ -77,6 +89,29 @@ public final class EdDSAPrivateKeyImpl
             throw new InvalidKeyException(ex);
         }
         checkLength(params);
+
+        final byte[] lh = this.h;
+        cleanable = CleanerFactory.cleaner().register(this,
+                                                      () -> Arrays.fill(lh,
+                                                                        (byte) 0x00));
+    }
+
+    /**
+     * Clears the internal copy of the key.
+     *
+     */
+    @Override
+    public void destroy() {
+        super.destroy();
```

Comment:
This is possible, due to the cleaner registered in PKCS8Key and should zero out its fields before doing the subclass's `destroy`.